src/common/id/Snowflake.h

```diff
@@ -0,0 +1,56 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+
+#ifndef COMMON_ID_SNOWFLAKE_H_
+#define COMMON_ID_SNOWFLAKE_H_
+
+#include "clients/meta/MetaClient.h"
+#include "common/base/Base.h"
+
+namespace nebula {
+class Snowflake {
+ public:
+  Snowflake() = default;
+
+  static void initWorkerId(meta::MetaClient* client) {
+    const std::string& ip = client->getLocalIp();
+    auto result = client->getWorkerId(ip).get();
+    if (!result.ok()) {
+      LOG(FATAL) << "Failed to get workerId from meta server";
+    }
+    workerId_ = result.value();
+  }
+
+  static void mockInitWorkerId(int32_t workerId) { workerId_ = workerId; }
```

Comment:
We don't need a interface like this i think, you could use `FREND_TEST` or `friend class` to make your test acces your private members.